jasmine/spec/inverted-index-test_spec.js

```diff
@@ -48,20 +67,39 @@ describe('Search index', () => {
   // ensure searchIndex can handle a varied number of search terms as arguments.
   it('should handle a varied number of search terms', () => {
     expect(index.searchIndex('books.json', ['a', 'alice'], 'book', 'me', ['help']))
-    .toEqual({ 'books.json':
-    { a: [0, 1],
-      alice: [0],
-      book: ['Not Found', 'Not Found'],
-      me: ['Not Found', 'Not Found'],
-      help: ['Not Found', 'Not Found'] } });
+      .toEqual({
+        'books.json': {
+          a: [0, 1],
+          alice: [0],
+          book: ['Not Found', 'Not Found'],
+          me: ['Not Found', 'Not Found'],
+          help: ['Not Found', 'Not Found'],
+        },
+      });
   });
 
-  /**
-    * ensure searchIndex goes through all indexed files if a filename is not passed, 
-    * i.e filename argument should be made optional
-  **/
+  // ensures if file is not specified, all files are searched
   it('should search all files if file is not specified', () => {
-    expect(index.searchIndex(undefined, 'alliance', 'alice', 'powerful', 'comical'))
-    .toEqual('books.json: alliance: 1,alice: 0,powerful: 1, test.json: comical: 1');
+    index.createIndex('test.json', test);
+    const results = [];
+    Object.keys(index.files).forEach((obj) => {
+      results.push(index.searchIndex(obj, 'alliance', 'alice', 'powerful', 'comical'));
```

Comment:
### Nitpick

`obj` should actually be filename, because that's exactly what it is, your naming is confusing, please rectify.